src/Network/Datadog/Monitor.hs

```diff
@@ -0,0 +1,388 @@
+                       , mSpec :: MonitorSpec
+                         -- ^ The specification from which this monitor can be
+                         -- re-created.
+                       } deriving (Eq)
+
+instance ToJSON Monitor where
+  toJSON monitor = Object $ Data.HashMap.union basemap newmap
+    where (Object basemap) = toJSON (mSpec monitor)
+          (Object newmap) = object ["id" .= mId monitor]
+
+instance FromJSON Monitor where
+  parseJSON (Object v) = modifyFailure ("Monitor: " ++ ) $
+                         Monitor <$> v .: "id" <*> parseJSON (Object v)
+  parseJSON a = modifyFailure ("Monitor: " ++) $ typeMismatch "Object" a
+
+
+-- | Create a new monitor in Datadog matching a specification.
+createMonitor :: Environment -> MonitorSpec -> IO Monitor
+createMonitor env monitorspec =
+  let path = "monitor"
+  in datadogHttp env path [] "POST" (Just $ encode monitorspec) >>=
+     decodeDatadog "createMonitor"
+
+
+-- | Load a monitor from Datadog by its ID.
+loadMonitor :: Environment -> MonitorId -> IO Monitor
+loadMonitor env monitorId =
+  let path = "monitor/" ++ show monitorId
+  in datadogHttp env path [] "GET" Nothing >>=
+     decodeDatadog "loadMonitor"
+
+
+-- | Sync a monitor with Datadog.
+--
+-- This must be called on any active monitors to apply the changes with Datadog
+-- itself; merely modifying a monitor locally is not enough to store the
+-- changes.
+--
+-- /Beware/: If a monitor has changed on the Datadog remote endpoint between
+-- the time it was copied locally and when this function is called, those
+-- changes already made remotely will be overwritten by this change.
+updateMonitor :: Environment -> MonitorId -> MonitorSpec -> IO Monitor
+updateMonitor env monitorId mspec =
+  let path = "monitor/" ++ show monitorId
+  in datadogHttp env path [] "PUT" (Just $ encode mspec) >>=
+     decodeDatadog "updateMonitor"
+
+
+-- | Delete a monitor from Datadog.
+--
+-- Note that once a monitor is deleted, it cannot be used locally anymore,
+-- however you can always create a new monitor using the deleted monitor's
+-- specification.
+deleteMonitor :: Environment -> MonitorId -> IO ()
+deleteMonitor env monitorId =
+  let path = "monitor/" ++ show monitorId
+  in void $ datadogHttp env path [] "DELETE" Nothing
+
+
+-- | Load monitors from Datadog.
+--
+-- This function takes a filter list argument, which should contain any tags
+-- the user wishes to filter on.  If the filter list is left empty, no filters
+-- will be applied.  The list of tags is ANDed together; if you specify more
+-- than one filter tag, only metrics which match all filter tags will be
+-- provided.
+loadMonitors :: Environment
+             -> [Tag]
+             -- ^ Tags on which to filter the results
+             -> IO [Monitor]
+loadMonitors env tags =
+  let path = "monitor"
+      query = [("tags", intercalate "," (map show tags)) | not (null tags)]
+  in datadogHttp env path query "GET" Nothing >>=
```

Comment:
Consider using the constants defined in Network.HTTP.Types.Method to replace `"GET"`, `"POST"`, `"PUT"`, and `"DELETE"` string literals.